### PR TITLE
Workaround for Cuda9.2 and GCC7 compilation errors

### PR DESCRIPTION
**Pull request by mingzhe09088**
Breaking out of #8338 

This PR is a workaround for a bug with CUDA9.2 + GCC7. 

Here is the error this PR fixed: 
.../pytorch/caffe2/operators/elementwise_ops.h: In constructor ‘caffe2::BinaryElementwiseWithArgsOp<InputTypes, Context, Functor, OutputTypeMap>::BinaryElementwiseWithArgsOp(const caffe2::OperatorDef&, caffe2::Workspace*)’:
.../pytorch/caffe2/operators/elementwise_ops.h:106:189: error: ‘GetSingleArgument<bool>’ is not a member of ‘caffe2::BinaryElementwiseWithArgsOp<InputTypes, Context, Functor, OutputTypeMap>’
   BinaryElementwiseWithArgsOp(const OperatorDef& operator_def, Workspace* ws)

### CODE REVIEW

**Review by orionr — COMMENTED** [review]
I don't think the Windows failure is caused by you, so this looks clean! Well done.

**Review by orionr — APPROVED** [review]

File: caffe2/core/operator.h [hunk 1]
```diff
@@ -420,8 +420,14 @@ class OperatorBase : public Observable<OperatorBase> {
 
 // OP_SINGLE_ARG provides a shorter initialization choice for initialization of
 // member variables for the class constructors.
+// This is a workaround for CUDA9.2 and GCC7
+#if defined(CUDART_VERSION) && CUDART_VERSION >= 9020 && __GNUC__ >= 7
+#define OP_SINGLE_ARG(type, name, variable, default)                           \
+  variable(this->template GetSingleArgument<type>(name, (default)))
```
Comment (ezyang) [inline caffe2/core/operator.h]: Is there any particular reason we couldn't just use this definition unconditionally? It seems like it should be valid for non CUDA9.2+GCC7 configs too.

Comment (ezyang) [reply caffe2/core/operator.h]: And indeed, in the rest of the patch you rewrite this unconditionally.

**Comment by ezyang** [comment]
Merge this ASAP, you can fix the nit later if you think it's a good idea.

**Pull request closed without merging**